shell/browser/node_service_host_impl.cc

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2022 Microsoft, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "shell/browser/node_service_host_impl.h"
+
+#include <utility>
+
+#include "shell/browser/api/electron_api_utility_process.h"
+
+namespace electron {
+
+NodeServiceHostImpl::NodeServiceHostImpl() = default;
+
+NodeServiceHostImpl::~NodeServiceHostImpl() = default;
+
+void NodeServiceHostImpl::BindReceiver(
+    mojo::PendingReceiver<node::mojom::NodeServiceHost> receiver) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  receivers_.Add(this, std::move(receiver));
+}
+
+void NodeServiceHostImpl::ReceivePostMessage(
+    base::ProcessId pid,
+    blink::TransferableMessage message) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  auto* utility_process_wrapper = GetAllUtilityProcessWrappers().Lookup(pid);
```

Comment:
I feel like the wrapper should be bound more tightly than just via pid... don't we already know which wrapper this is being delivered to from the pipe's identity?

Reply:
the interface name is not always unique, given this is a `GenericReceiver` on the browser process I was not sure how to target a particular wrapper without additional data sent across the pipe and I chose pid as the additional uniqueness for now. Is there a different identity you are referring to ?

Reply:
I'm not sure I follow; there ought to be a way to have a separate host object for each utility process. Happy to look through the code with you at some point if it's not clear how to make that happen!

Reply:
Misunderstanding on my part about the API, will resolve via https://github.com/electron/electron/pull/34980#discussion_r983258198